Library/SPIClient.m

```diff
@@ -1523,6 +1524,26 @@ - (SPIMessageSuccessState)gltMatch:(SPIGetLastTransactionResponse *)gltResponse
     return gltResponse.getSuccessState;
 }
 
+- (SPIMessageSuccessState)gltMatch:(SPIGetLastTransactionResponse *)gltResponse
+                    expectedAmount:(NSInteger)expectedAmount
+                       requestDate:(NSDate *)requestDate
+                          posRefId:(NSString *)posRefId {
+    SPILog(@"GLT check: posRefId=%@->%@}", posRefId, [gltResponse getPosRefId]);
+    
+    NSDate *gltBankDate = [[NSDateFormatter dateFormaterWithFormatter:@"ddMMyyyyHHmmss"] dateFromString:[gltResponse getBankDateTimeString]];
+    BOOL compare = [requestDate compare:gltBankDate];
+    
+    if (![posRefId isEqualToString:gltResponse.getPosRefId]) {
+        return SPIMessageSuccessStateUnknown;
+    }
+    
+    if ([[gltResponse getTxType] isEqual: @"PURCHASE"] && [gltResponse getBankNonCashAmount] != expectedAmount && compare > 0) {
```

Comment:
Is this case sensitive?

Reply:
Yes, it is case sensitive. I tested it like this and it works fine.

Reply:
Please convert toUpper thanks